### PR TITLE
Fix Review

### PR DESCRIPTION
**Pull request by sherlock-admin4**
Fix Review of 

Repo: `saffron-finance/lido-fiv`
Commit Hash: `bf79565e346bb410df260f51483c75ca00ba25e2`

### CODE REVIEW

File: lido-fiv/contracts/LidoVault.sol [hunk 1]
```diff
@@ -47,7 +47,7 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
   uint256 public appliedProtocolFee;
 
   /// @notice ETH amount that tracks the total Saffron protocol fee from variable withdrawal staking earnings while the vault is still ongoing
-  uint256 public totalProtocolFee;
+  uint256 public ongoingProtocolFeeInShares;
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:50]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 2]
```diff
@@ -123,6 +123,9 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
   /// @notice Amount of ETH used to cover the returning of fixed user's initial principal
   uint256 public vaultEndedFixedDepositsFunds;
 
+    /// @notice Amount of earnings in ETH from Lido Staking after the vault has ended that was already withdrawn
+  uint256 public vaultEndedWithdrawnStakingEarnings;
+
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:128]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 3]
```diff
@@ -176,7 +179,7 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
   mapping(address => uint256) public variableToWithdrawnStakingEarningsInShares;
 
   /// @notice Mapping from variable side user addresses to their total payed protocol fee amount right after their Lido withdrawal request is submitted
-  mapping(address => uint256) public variableToWithdrawnProtocolFee;
+  mapping(address => uint256) public variableToWithdrawnProtocolFeeInShares;
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:182]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 4]
```diff
@@ -521,24 +524,26 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
         // staking earnings have accumulated on Lido
-        if (lidoStETHBalance > fixedETHDeposits + minStETHWithdrawalAmount()) {
+        if (lidoStETHBalance > fixedETHDeposits + MIN_STETH_WITHDRAWAL_AMOUNT) {
           uint256 currentStakes = stakingShares();
           (uint256 currentState, uint256 ethAmountOwed) = calculateVariableWithdrawState(
             (lidoStETHBalance.mulDiv(currentStakes + withdrawnStakingEarningsInStakes, currentStakes) - fixedETHDeposits),
             variableToWithdrawnStakingEarningsInShares[msg.sender].mulDiv(lidoStETHBalance, currentStakes)
           );
-          if (ethAmountOwed >= minStETHWithdrawalAmount()) {
+          ethAmountOwed = Math.min(ethAmountOwed, lidoStETHBalance - fixedETHDeposits);
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:529]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 5]
```diff
@@ -521,24 +524,26 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+          if (ethAmountOwed >= MIN_STETH_WITHDRAWAL_AMOUNT) {
             // estimate protocol fee and update total - will actually be applied on withdraw finalization
             uint256 protocolFee = ethAmountOwed.mulDiv(protocolFeeBps, 10000);
-            totalProtocolFee += protocolFee;
+            uint256 protocolFeeInShares = lido.getSharesByPooledEth(protocolFee);
+            ongoingProtocolFeeInShares += protocolFeeInShares;
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:534]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 6]
```diff
@@ -521,24 +524,26 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
             uint256 stakesAmountOwed = lido.getSharesByPooledEth(ethAmountOwed);
 
             withdrawnStakingEarnings += ethAmountOwed - protocolFee;
             withdrawnStakingEarningsInStakes += stakesAmountOwed;
 
             variableToWithdrawnStakingEarnings[msg.sender] += ethAmountOwed - protocolFee;
             variableToWithdrawnStakingEarningsInShares[msg.sender] += stakesAmountOwed;
-            variableToWithdrawnProtocolFee[msg.sender] += protocolFee;
+            variableToWithdrawnProtocolFeeInShares[msg.sender] += protocolFeeInShares;
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:542]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 7]
```diff
@@ -573,6 +578,19 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+    require(bearerBalance > 0, "NBT");
+    if (feeEarnings > 0) {
+      uint256 feeEarningsShare = calculateVariableFeeEarningsShare();
+      if (feeEarningsShare > 0) {
+        transferWithdrawnFunds(msg.sender, feeEarningsShare);
+        emit VariableFundsWithdrawn(feeEarningsShare, msg.sender, true, false);
+      }
+    }
+  }
+
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:589]: Change related to https://github.com/saffron-finance/lido-fiv/pull/30. 

Fixes issues: [#105](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/105)

File: lido-fiv/contracts/LidoVault.sol [hunk 8]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
-        variableToWithdrawnStakingEarnings[msg.sender] = currentState;
+      if (vaultEndingETHBalance >= MIN_STETH_WITHDRAWAL_AMOUNT) {
+        uint256 totalEarnings = Math.max(withdrawnStakingEarnings,
+             vaultEndingETHBalance.mulDiv(withdrawnStakingEarningsInStakes - ongoingProtocolFeeInShares,vaultEndingStakesAmount)) + vaultEndedStakingEarnings;
+        if (totalEarnings > 0) {
+          (uint256 currentState, uint256 stakingEarningsShare) = calculateVariableWithdrawState(
+            totalEarnings,
+            Math.max((variableToWithdrawnStakingEarningsInShares[msg.sender] - variableToWithdrawnProtocolFeeInShares[msg.sender]).mulDiv(vaultEndingETHBalance, vaultEndingStakesAmount),
+            variableToWithdrawnStakingEarnings[msg.sender])
+          );
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:810]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 9]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+          stakingShareAmount = Math.min(stakingEarningsShare, vaultEndedStakingEarnings - vaultEndedWithdrawnStakingEarnings);
+          vaultEndedWithdrawnStakingEarnings += stakingShareAmount;
+          variableToWithdrawnStakingEarningsInShares[msg.sender] += stakingShareAmount.mulDiv(vaultEndingStakesAmount,vaultEndingETHBalance);
+          variableToWithdrawnStakingEarnings[msg.sender] += stakingShareAmount;
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:814]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 10]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+        }
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:815]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 11]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
       }
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:817]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 12]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
             await lidoVault.connect(addr2).withdraw(SIDE.VARIABLE)
             expect(await lidoVault.vaultEndedWithdrawalsFinalized()).to.equal(false)
 
+            await lidoVault.connect(addr1).finalizeVaultEndedWithdrawals(SIDE.FIXED)
```
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:6785]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 13]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
...
             balanceBefore = await ethers.provider.getBalance(addr2)
+            console.log(await lidoVault.withdrawnStakingEarnings() , await lidoVault.vaultEndedStakingEarnings())
+            console.log(await lidoVault.vaultEndingETHBalance() * (await lidoVault.withdrawnStakingEarningsInStakes() -
+            await lidoVault.ongoingProtocolFeeInShares())/ await lidoVault.vaultEndingStakesAmount())
```
Comment (sherlock-admin3) [inline lido-fiv/test/1.LidoVault.test.ts:6789]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 14]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
...
             const receipt = await (
               await lidoVault.connect(addr2).finalizeVaultEndedWithdrawals(SIDE.VARIABLE)
             ).wait()
             const gasFees = calculateGasFees(receipt)
             const balanceAfter = await ethers.provider.getBalance(addr2)
 
             expect(await lidoVault.vaultEndedWithdrawalsFinalized()).to.equal(true)
+            expect(await lidoVault.variableBearerToken(addr2.address)).to.equal(0)
             expect(balanceAfter).to.equal(balanceBefore - gasFees)
```
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6798]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 15]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
...
+
+            const balanceAddr3Before = await ethers.provider.getBalance(addr3)
+
+            await expect(
+              lidoVault.connect(addr3).withdraw(SIDE.VARIABLE)
+            ).not.to.be.revertedWith('ETF')
+            await lidoVault.connect(addr3).finalizeVaultOngoingVariableWithdrawals()
+            console.log(balanceAddr3Before -  await ethers.provider.getBalance(addr3))
+
           })
```
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:6896]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 16]
```diff
@@ -8033,8 +8134,8 @@ describe('LidoVault', function () {
                 ).wait()
                 const gasFees2 = calculateGasFees(receipt2)
                 const balanceAfter2 = await ethers.provider.getBalance(addr2)
-                expect(balanceAfter2 - balanceBefore2).to.equal(
-                  earningsShare2 - gasFees2
+                expect(balanceAfter2 - balanceBefore2).to.equalWithTolerance(
+                  earningsShare2 - gasFees2, 10
```
Comment (sherlock-admin3) [inline lido-fiv/test/1.LidoVault.test.ts:8138]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 1 above)
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:50]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 2 above)
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:128]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 3 above)
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:182]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 17]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
 
       // Return proportional share of both earnings to caller
       uint256 stakingShareAmount = 0;
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:801]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 18]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
-        stakingShareAmount = stakingEarningsShare;
-        variableToWithdrawnStakingEarningsInShares[msg.sender] = currentState.mulDiv(vaultEndingStakesAmount,vaultEndingETHBalance);
-        variableToWithdrawnStakingEarnings[msg.sender] = currentState;
+      if (vaultEndingETHBalance >= MIN_STETH_WITHDRAWAL_AMOUNT) {
+        uint256 totalEarnings = Math.max(withdrawnStakingEarnings,
+             vaultEndingETHBalance.mulDiv(withdrawnStakingEarningsInStakes - ongoingProtocolFeeInShares,vaultEndingStakesAmount)) + vaultEndedStakingEarnings;
+        if (totalEarnings > 0) {
+          (uint256 currentState, uint256 stakingEarningsShare) = calculateVariableWithdrawState(
+            totalEarnings,
+            Math.max((variableToWithdrawnStakingEarningsInShares[msg.sender] - variableToWithdrawnProtocolFeeInShares[msg.sender]).mulDiv(vaultEndingETHBalance, vaultEndingStakesAmount),
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:808]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 19]
```diff
@@ -771,19 +791,23 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+            variableToWithdrawnStakingEarnings[msg.sender])
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:809]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 9 above)
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:814]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 10 above)
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:815]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 12 above)
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6785]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 13 above)
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:6789]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 14 above)
Comment (sherlock-admin3) [inline lido-fiv/test/1.LidoVault.test.ts:6798]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 15 above)
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6896]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 16 above)
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:8138]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

**Pull request closed without merging**